Fix server installation and unwanted file deletion

DrParadox7 opened the pull request:
Server will now close on successful installation or error to properly load up mods at the next start.

Prevents MCIL from deleting what it considers it to be "empty files" at every load which would often have it deleting important files. Now it shall only do so prior to an installation.

DrParadox7 committed:
Restart server on successful install

DrParadox7 committed:
Disable the recurrent deletion of empty files

Prevents MCIL from constantly deleting small files from the mod folder at every launch.

It will now only do so if there is a valid "pack.mcinstance" that needs to be installed.

This prevents the unintentional deletion of small files such as Ruins templates or EJML-Core from IC2.

DrParadox7 committed:
Update Version to 2.5

DrParadox7 committed:
Server Additions

-Log will print the installation progress server-side
-Reassure users that a server restart is necessary and not a crash.

DrParadox7 commented:
Added progress count server side so users can see the installation progress:
**Before** *(MCIL 2.4)*:
![image](https://user-images.githubusercontent.com/47131096/206876170-54506307-f760-484a-8fc9-b292f3893c9f.png)

**After** *(MCIL 2.5)*:
![image](https://user-images.githubusercontent.com/47131096/206876085-99c6e968-d198-4b5a-b62a-ac1ed05cfb2b.png)

Also the log will now print `"The server will now restart to complete the setup."` when the installation process is complete to prevent crash confusion:

DrParadox7 committed:
Added Localized Overrides

-Added localized overrides depending on what side MCIL is being run at:

[overrides]: Applies to both Servers and Clients
[client_override]: Client-only files
[server_overrides]: Server-only files

HRudyPlayZ commented:
@DrParadox7 Very good work !

About the file deletion patch, i now see what you were talking about. The mod can indeed cause issues by deleting files too early, i thought about it at the time but i forgot this issue.
Your patch can work indeed (i'll see what i can do more in 3.0 as it's gonna be a different beast), but it would break the update checker, i'd change the code to be something like this instead:

```java
        String[] list = FileHelper.listDirectory("mods", true);
        for (String s : list) {
            long size = -1;
            try {
                size = Files.size(Paths.get("mods" + File.separator + s));
            }
            catch (Exception ignore) {}

            if (size == 0 && (FileHelper.exists(Config.configFolder + "pack.mcinstance") || s.startsWith("mcinstanceloader"))) FileHelper.delete("mods" + File.separator + s);
        }
```

The restart patch is perfect pretty much, i can't believe i forgot it xD

The server additions patch is also pretty good, i'd just remove the pipe separator (|) as i feel it makes the line slightly harder to read.

Same thing for the localized overrides. I haven't tested it but from what i see it shouldn't cause any issue. Could you rename the folders to `client-overrides` and `server-overrides` instead of using underscores (_)? I feel this would work better as a file name.

If you make those changes, i'll merge the PR and release 2.5 quickly after :D

DrParadox7 committed:
PR Requests

-Refactors deletion module to only fire if MCIL is in the process of either installing a pack or updating itself.

-Renamed localized overrides folders to use "-" over "_".

-Removed "|" from server download logs and renamed it "mods" to" resources" for accuracy.

-Reworded server restart message from using "now" to "soon" since the restart is not immediate.

DrParadox7 commented:
Done. 
I did also reworded some of the comments/logs to be more accurate.

DrParadox7 committed:
Properly fixed unintended deletions

Patches MCIL from deleting folders it is not supposed to by erroneously detecting it as empty.

Added verbose info to debug similar occurrences in the future.

DrParadox7 committed:
Cleanup

Clean up code for readability

DrParadox7 committed:
Fix wrong statement used
